File: src/at/fhooe/mc/emg/tools/peak/PeakDetectionTool.java
package at.fhooe.mc.emg.tools.peak;

import at.fhooe.mc.emg.core.EmgController;
import at.fhooe.mc.emg.tools.Tool;

import javax.swing.*;

/**
 * Author:  Mescht
 * Date:    04.07.2017
 */
public class PeakDetectionTool implements Tool {

    public PeakDetectionTool() {

    }

    @Override
    public String getName() {
        return "Peak detection";
    }

    @Override
    public void start(EmgController controller) {

        JFrame frame = new JFrame();
        frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
        frame.setTitle(getName());
        frame.setContentPane(new JPanel());
        frame.setBounds(600, 200, 400, 300);
        frame.setVisible(true);
    }
}

File: library/src/at/fhooe/mc/emg/tools/peak/PeakDetectionTool.kt
package at.fhooe.mc.emg.tools.peak

import at.fhooe.mc.emg.core.EmgController
import at.fhooe.mc.emg.tools.Tool
import javax.swing.JFrame
import javax.swing.JPanel
import javax.swing.WindowConstants

/**
 * Author:  Mescht
 * Date:    04.07.2017
 */
class PeakDetectionTool : Tool {

    override val name: String
        get() = "Peak detection"

    override fun start(controller: EmgController) {

        val frame = JFrame()
        frame.defaultCloseOperation = WindowConstants.DISPOSE_ON_CLOSE
        frame.title = name
        frame.contentPane = JPanel()
        frame.setBounds(600, 200, 400, 300)
        frame.isVisible = true
    }
}
